graphula-core/src/Graphula.hs

```diff
@@ -190,16 +179,16 @@ instance MonadUnliftIO m => MonadUnliftIO (GraphulaIdempotentT m) where
 instance MonadTrans GraphulaIdempotentT where
   lift = GraphulaIdempotentT . lift
 
-instance (Monad m, MonadIO m, MonadGraphulaFrontend m) => MonadGraphulaFrontend (GraphulaIdempotentT m) where
-  type NodeConstraint (GraphulaIdempotentT m) = NodeConstraint m
-  type Node (GraphulaIdempotentT m) = Node m
+  {-
+  TODO figure out how to support tracking keys.
```

Comment:
Not having `MonadGraphulaFrontend` breaks our ability to handle idempotent graphs, since we cannot overload `insert` based on the the context.

Possibly we want to key `MonadGraphulaFrotend` and redefine `insert` as
```
insert :: Maybe (Key a) -> a -> m (Entity a)
```

Reply:
Haha jinx